src/rules/default-import-match-filename.js

```diff
@@ -0,0 +1,120 @@
+import docsUrl from '../docsUrl'
+import isStaticRequire from '../core/staticRequire'
+
+/**
+ * @param {string} filename
+ * @returns {string}
+ */
+function removeExtension(filename) {
+  return filename.replace(/\.[^.]+$/, '')
+}
+
+/**
+ * @param {string} filename
+ * @returns {string}
+ */
+function normalizeFilename(filename) {
+  return filename.replace(/[-_.]/g, '').toLowerCase()
+}
+
+/**
+ * Test if local name matches filename.
+ * @param {string} localName
+ * @param {string} filename
+ * @returns {boolean}
+ */
+function isCompatible(localName, filename) {
+  const normalizedLocalName = localName.replace(/_/g, '').toLowerCase()
+
+  return (
+    normalizedLocalName === normalizeFilename(filename) ||
+    normalizedLocalName === normalizeFilename(removeExtension(filename))
+  )
+}
+
+/**
+ * Test if path starts with "./" or "../".
+ * @param {string} path
+ * @returns {boolean}
+ */
+function isLocalModule(path) {
+  return /^(\.\/|\.\.\/)/.test(path)
+}
+
+/**
+ * Get filename from a path.
+ * @param {string} path
+ * @returns {string | undefined}
+ */
+function getFilename(path) {
+  if (!isLocalModule(path)) return undefined
+  const [, filename] = /\/([^/]*)$/.exec(path)
+  if (filename === '' || filename === '.' || filename === '..') {
```

Comment:
with either `.` or `../`, i would expect the import name to be "the current directory name" (which would prefer `package.json`'s name, if there was a `package.json` in the current directory).

Reply:
Done. File IO scares me.

Reply:
I think this use case brings lots of complexity (if then else, file IO, and reading `package.json`) with little benefit. Using `require('..')` or `require('../..')` happens rare enough and is an anti pattern IMO. I would like to ignore this case.

And a hairy case: `import src from '..';`, when `..` resolves to `src`.

Reply:
In no way is it an antipattern; and it happens in every `test` dir in all my projects.

The recursive case is just as hairy with `from '.'`, and both need to be handled.

Reply:
This should be resolved.